src/MoBi.Presentation/MenusAndBars/ContextMenus/MultipleBuildingBlockNodeContextMenuFactory.cs

```diff
@@ -35,10 +40,15 @@ public MultipleBuildingBlockNodeContextMenu(IReadOnlyList<IBuildingBlock> buildi
 
       protected override IEnumerable<IMenuBarItem> AllMenuItemsFor(IReadOnlyList<IBuildingBlock> buildingBlocks, IMoBiContext context)
       {
-         if (buildingBlocks.Count == 2)
+         if (buildingBlocks.Count == 2 && (buildingBlocks[0].GetType() == buildingBlocks[1].GetType()))
             yield return ComparisonCommonContextMenuItems.CompareObjectsMenu(buildingBlocks, context, _container);
 
          yield return ObjectBaseCommonContextMenuItems.AddToJournal(buildingBlocks, _container);
+
+         if (buildingBlocks.Count > 1)
+            yield return CreateMenuButton.WithCaption(AppConstants.MenuNames.Delete)
+               .WithCommandFor<RemoveMultipleBuildingBlocksUICommand, IReadOnlyList<IBuildingBlock>>(buildingBlocks, _container)
+               .WithIcon(ApplicationIcons.Delete);
```

Comment:
If BBS count is = 1: I don't get the delete menu? Or is it added elsewhere?

Reply:
Yeah it's added elsewhere. This CM is only when you have multiple selected. For that reason, it probably does not need to be checked.

Reply:
ok